### PR TITLE
Produce OpenTelemetry metrics

### PR DESCRIPTION
**Pull request by edigaryev**
Resolves https://github.com/cirruslabs/orchard/issues/178.

### CODE REVIEW

File: DeploymentGuide.md [hunk 1]
```diff
@@ -126,13 +126,34 @@ orchard context default production
...
 Please refer a [separate repository](https://github.com/cirruslabs/ansible-orchard) where we prepared a basic
 Ansible playbook for convenient setup.
+
+## Observability
+
+Both the controller and worker produce some useful OpenTelemetry metrics.
+
+By default, the telemetry is sent to https://localhost:4317 using the gRPC protocol and to http://localhost:4318 using the HTTP protocol.
+
+You can override this by setting the [standard OpenTelemetry environment variable](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/) `OTEL_EXPORTER_OTLP_ENDPOINT`.
```
Comment (fkorotkov) [inline DeploymentGuide.md:159]:
```suggestion
You can override this by setting the [standard OpenTelemetry environment variable](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/) `OTEL_EXPORTER_OTLP_ENDPOINT`.

Please refer to [OTEL Collector documentation](https://opentelemetry.io/docs/collector/) for instruction on how to setup a sidecar for the metrics collections or find out if your SaaS monitoring has an available OTEL endpoint (see [Honeycomb](https://docs.honeycomb.io/send-data/opentelemetry/) as an example).
```

File: internal/controller/controller.go [hunk 2]
```diff
@@ -239,3 +251,87 @@ func (controller *Controller) SSHAddress() (string, bool) {
...
+
+				groups := lo.CountValuesBy(vms, func(vm v1.VM) Key {
+					return Key{
+						Worker: vm.Worker,
+						Status: vm.Status,
+					}
+				})
+
+				for key, count := range groups {
+					observer.Observe(int64(count), metric.WithAttributes(
```
Comment (fkorotkov) [inline internal/controller/controller.go:277]: Should we observe `0`s for missing statuses?

File: internal/controller/controller.go [hunk 3]
```diff
@@ -239,3 +251,87 @@ func (controller *Controller) SSHAddress() (string, bool) {
...
+		}),
+	)
+	if err != nil {
+		return err
+	}
+
+	_, err = opentelemetry.DefaultMeter.Int64ObservableGauge("org.cirruslabs.orchard.controller.worker_resource",
+		metric.WithInt64Callback(func(ctx context.Context, observer metric.Int64Observer) error {
+			return controller.store.View(func(txn storepkg.Transaction) error {
+				workers, err := txn.ListWorkers()
```
Comment (fkorotkov) [inline internal/controller/controller.go:294]: Same question when there are no workers

File: internal/controller/controller.go [hunk 4]
```diff
@@ -239,3 +251,87 @@ func (controller *Controller) SSHAddress() (string, bool) {
 
 	return controller.sshServer.Address(), true
 }
+
+func (controller *Controller) initializeMetrics() error {
```
Comment (fkorotkov) [inline internal/controller/controller.go:255]: I think having online/offline workers counter is also beneficial.

File: DeploymentGuide.md [hunk 5]
```diff
@@ -126,13 +126,34 @@ orchard context default production
...
+
+### Automation
 
 If you have a set of machines that you want to use as Orchard Workers, you can use Ansible to configure them.
 Please refer a [separate repository](https://github.com/cirruslabs/ansible-orchard) where we prepared a basic
 Ansible playbook for convenient setup.
+
+## Observability
+
+Both the controller and worker produce some useful OpenTelemetry metrics.
```
Comment (fkorotkov) [inline DeploymentGuide.md:155]:
```suggestion
Both the controller and worker produce some useful OpenTelemetry metrics. Metrics are scoped with `org.cirruslabs.orchard` prefix and include information about resource utilization, statuses or workers, scheduling/pull time and many more.
```

File: internal/controller/controller.go (hunk 4 above)
Comment (edigaryev) [reply internal/controller/controller.go:256]: Implemented in https://github.com/cirruslabs/orchard/pull/185/commits/ea0e1f70eb843fc146b6b92a074a8f4dde6e1221.

**Review by fkorotkov — APPROVED** [review]

**Pull request merged**